### PR TITLE
[WIP] VB Feature (PROTOTYPE) EnumFlagsExpression.

### PR DESCRIPTION
**Pull request by AdamSpeight2008**
Initial work on a prototype implementation of EnumFlagsExpression. ([VBLang Issue](https://github.com/dotnet/vblang/issues/228))
- [x] Parser
- [x] Binder
- [x] Lowering ( *ignoring assert failures*)
  
flagsenumexpression_error
flagsenumexpression

### CODE REVIEW

File: src/Compilers/VisualBasic/Portable/BoundTree/BoundNodes.xml [hunk 1]
```diff
@@ -691,6 +691,15 @@
     <Field Name="AccessKind" Type="LateBoundAccessKind"/>
   </Node>
 
+  <Node Name="BoundEnumFlagExpression" Base="BoundExpression" HasValidate="true">
```
Comment (AdamSpeight2008) [inline src/Compilers/VisualBasic/Portable/BoundTree/BoundNodes.xml]: @AnthonyDGreen 
What do think the requires shape for this node should be?

**Comment by AdamSpeight2008** [comment]
@AnthonyDGreen 
How do I bind to a enum value?

**Comment by AdamSpeight2008** [comment]
Would a member of @roslyn-compiler offering a quick review just see I'm going roughly in the right direction.

**Comment by jaredpar** [comment]
PRs for language features go against our [contribution guidelines](https://github.com/dotnet/roslyn/blob/master/CONTRIBUTING.md)

> ***DO NOT*** submit language features as PRs to this repo first, or they will likely be declined.

Please follow our contribution guidelines for language features.

**Comment by AlekseyTs** [comment]
Also, I believe the feature hasn't been approved by LDM yet.

**Comment by AdamSpeight2008** [comment]
@AlekseyTs  It is a prototype implementation for the proposal in VB Lang PR.

**Comment by AlekseyTs** [comment]
A validation from LDM is a prerequisite for any prototyping, which involves (even as code reviewers) team members. From  [contribution guidelines](https://github.com/dotnet/roslyn/blob/master/CONTRIBUTING.md):
Once a feature is **championed** and **validated by LDM** a developer will be assigned to help begin a prototype on this repo inside a feature branch.

**Comment by AdamSpeight2008** [comment]
I've got a working [Proof Of Concept](https://github.com/AdamSpeight2008/roslyn-AdamSpeight2008/tree/EnumFlagExpression).

**Pull request closed without merging**